homeassistant/components/imgw_pib/sensor.py

```diff
@@ -28,14 +30,36 @@
 PARALLEL_UPDATES = 0
 
 
+def gen_attributes(data: HydrologicalData) -> dict[str, Any] | None:
+    """Generate attributes for the alert entity."""
+    if data.alert.value == NO_ALERT:
+        return None
+
+    return {
+        "level": data.alert.level,
+        "probability": data.alert.probability,
+        "valid_from": data.alert.valid_from,
+        "valid_to": data.alert.valid_to,
```

Comment:
I know we should avoid using extra attributes, but in this case, it makes sense. This additional data explains the state of the entity.
Besides, additional attributes can be used using Tile card to show the alert and its level

<img width="518" height="73" alt="Zrzut ekranu 2025-07-15 222816" src="https://github.com/user-attachments/assets/a0e09bab-3749-4045-b1df-77dd90948b7c" />